internal/accounts/create-interactive.go

```diff
@@ -0,0 +1,268 @@
+		"%s New account created with address %s and name %s on %s network.\n",
+		output.SuccessEmoji(),
+		output.Bold(fmt.Sprintf("0x%s", account.Address().String())),
+		output.Bold(name),
+		output.Bold(networkName)),
+	)
+
+	state.Accounts().AddOrUpdate(account)
+	err = state.SaveDefault()
+	if err != nil {
+		return err
+	}
+
+	items := []string{
+		"Here’s a summary of all the actions that were taken",
+		fmt.Sprintf("Added the new account to %s.", output.Bold("flow.json")),
+	}
+	if selectedNetwork != config.DefaultEmulatorNetwork() {
+		items = append(items,
+			fmt.Sprintf("Saved the private key to %s.", output.Bold(privateFile)),
+			fmt.Sprintf("Added %s to %s.", output.Bold(privateFile), output.Bold(".gitignore")),
+		)
+	}
+	outputList(log, items, false)
+
+	return nil
+}
+
+// createNetworkAccount using the account creation API and return the newly created account address.
+func createNetworkAccount(
+	state *flowkit.State,
+	services *services.Services,
+	name string,
+	key crypto.PrivateKey,
+	privateFile string,
+	network config.Network,
+) (*flowkit.Account, error) {
+	networkAccount := &lilicoAccount{
+		PublicKey: strings.TrimPrefix(key.PublicKey().String(), "0x"),
+	}
+
+	id, err := networkAccount.create(network.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	result, err := getAccountCreationResult(services, id)
+	if err != nil {
+		return nil, err
+	}
+
+	events := flowkit.EventsFromTransaction(result)
+	address := events.GetCreatedAddresses()
+	if len(address) == 0 {
+		return nil, fmt.Errorf("account creation error")
+	}
+
+	err = util.AddToGitIgnore(privateFile, state.ReaderWriter())
+	if err != nil {
+		return nil, err
+	}
+
+	err = state.ReaderWriter().WriteFile(privateFile, []byte(key.String()), os.FileMode(0644))
+	if err != nil {
+		return nil, fmt.Errorf("failed saving private key: %w", err)
+	}
+
+	return flowkit.NewAccount(name).SetAddress(*address[0]).SetKey(
+		flowkit.NewFileAccountKey(privateFile, 0, crypto.ECDSA_P256, crypto.SHA3_256),
+	), nil
+}
+
+func createEmulatorAccount(
+	state *flowkit.State,
+	service *services.Services,
+	name string,
+	key crypto.PrivateKey,
+) (*flowkit.Account, error) {
+	signer, err := state.EmulatorServiceAccount()
+	if err != nil {
+		return nil, err
+	}
+
+	networkAccount, err := service.Accounts.Create(
+		signer,
+		[]crypto.PublicKey{key.PublicKey()},
+		[]int{flow.AccountKeyWeightThreshold},
+		[]crypto.SignatureAlgorithm{crypto.ECDSA_P256},
+		[]crypto.HashAlgorithm{crypto.SHA3_256},
+		nil,
+	)
+
+	return flowkit.NewAccount(name).SetAddress(networkAccount.Address).SetKey(
+		flowkit.NewHexAccountKeyFromPrivateKey(0, crypto.SHA3_256, key),
+	), nil
+}
+
+func getAccountCreationResult(services *services.Services, id flow.Identifier) (*flow.TransactionResult, error) {
+	_, result, err := services.Transactions.GetStatus(id, true)
+	if err != nil {
+		if status.Code(err) == codes.NotFound { // if transaction not yet propagated, wait for it
+			time.Sleep(1 * time.Second)
```

Comment:
Would be nice to use a wait group or future/ channel to know when a transaction has been propagated. But if this doesn't come up much then it's okay.

Reply:
hmm, what do you mean? 
So the issue here is that when we use the account creation API they send us back the transaction ID immediately since that can be calculated offchain and we don't know when it will actually be propagated to network, that we don't have actual control over since we are not the ones who submit it, so it wouldn't be possible to have like a back channel to indicate that. A good idea in that case. The only thing we could is to monitor events and then as soon as the event with that account comes we know it was created but that has different downsides and in this case it only maybe takes one or two seconds to get there.